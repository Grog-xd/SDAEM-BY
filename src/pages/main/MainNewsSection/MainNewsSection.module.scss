@import "src/style/mixins";

.newsSection{
  display: flex;
  margin-bottom: 109px;
  justify-content: space-between;
  .textBlock{
    width: 48vw;
    .header{
      strong{
        background: linear-gradient(90deg, #9D94FF 0%, #6B50E9 94.5%);
        -webkit-background-clip: text;
        -webkit-text-fill-color: transparent;
        @include text(14px, 600, 17px, #000);
      }
      p{
        margin-top: 15px;
        @include text(25px, 700, 30px, #1E2123);
      }
    }
    .body{
      margin-top: 33px;
      position: relative;
      display: flex;
      .yellowPoints{
        position: absolute;
        z-index: 2;
        bottom: -20px;
        left: 37%;
      }
      .bodyImg{
        filter: drop-shadow(0px 15px 30px rgba(39, 95, 158, 0.1));
        border-radius: 20px;
      }
      .bodyText{
        margin-left: 24px;
        p{
          @include text(14px, 600, 21px, #1E2123);
          &:first-child{
            margin-bottom: 33px;
          }
          b{
            line-height: 24px;
          }
        }

      }
    }
    .footer{
      @include text(16px, 400, 21px, #1E2123);
      margin-top: 31px;
    }
  }
  .newsBlock{
    background: #FFFEFE;
    box-shadow: 0 20px 40px rgba(135, 124, 202, 0.15);
    border-radius: 10px;
    padding: 33px 23px;
    display: flex;
    flex-direction: column;
    margin-left: 30px;
    .title{
      margin-bottom: 16px;
      @include text(25px, 700, 30px, #1E2123);
    }
    .post{
      margin-top: 10px;
      padding-bottom: 10px;
      border-bottom: 1px solid rgba(102, 78, 249, 0.1);
      &:last-of-type{
        border: none;
      }
      a{
        @include text(14px, 600, 17px, #1E2123);
        &:hover{
          color: #664EF9;
        }
      }
    }
    .toNewsLink{
      display: flex;
      align-items: center;
      margin-top: 22px;
      @include text(14px, 700, 17px, #664EF9);
      svg{
        margin-left: 12px;
      }
    }

  }
}
