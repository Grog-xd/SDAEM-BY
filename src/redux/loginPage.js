import {createSlice} from '@reduxjs/toolkit';

import avatar from '../assets/img/avatar.png'

const loginPage = createSlice({
    name: 'login',
    initialState:{
        // bookMarkActive:false,
        isAuth: false,
        profilesArr:[{name: 'dmitriy', avatar:'', phone:'', city:'', email: 'vladimir6234@tut.by', password: '12345', viber: '', whatsUpp:''},{name: 'test', avatar:avatar, phone:'+test', city:'test', email: 'test@tut.by', password: 'test1', viber: 'test', whatsUpp:'test'}],
        profile: {},
    },
    reducers:{
        // setBookMarkActive(state){
        //     state.bookMarkActive = !state.bookMarkActive
        // },
        enter(state, action){
            state.isAuth = true
            state.profile = action.payload
        },
        exit(state){
            state.isAuth = false
            localStorage.clear()
        },
        registerUser(state, action){
            state.profilesArr = [...state.profilesArr, action.payload]
        },
    },
})

export default loginPage.reducer
export const {exit, enter, setBookMarkActive, registerUser} = loginPage.actions

