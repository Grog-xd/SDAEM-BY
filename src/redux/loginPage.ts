import {createSlice, PayloadAction} from '@reduxjs/toolkit';

import avatar from '../assets/img/avatar.png'
import {profilesArr} from '../types/types';

const loginPage = createSlice({
    name: 'login',
    initialState:{
        // bookMarkActive:false,
        isAuth: false,
        profilesArr:[{name: 'dmitriy', avatar:'', phone:'', city:'', email: 'vladimir6234@tut.by', password: '12345', viber: '', whatsUpp:''},{name: 'test', avatar:avatar, phone:'+test', city:'test', email: 'test@tut.by', password: 'test1', viber: 'test', whatsUpp:'test'}],
        profile: {},
    },
    reducers:{
        // setBookMarkActive(state){
        //     state.bookMarkActive = !state.bookMarkActive
        // },
        enter(state, action:PayloadAction<profilesArr>){
            state.isAuth = true
            state.profile = action.payload
        },
        exit(state){
            state.isAuth = false
            localStorage.clear()
        },
        registerUser(state, action:PayloadAction<any>){
            state.profilesArr = [...state.profilesArr, action.payload]
        },
    },
})

export default loginPage.reducer
export const {exit, enter, registerUser} = loginPage.actions

