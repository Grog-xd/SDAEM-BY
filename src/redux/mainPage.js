import {createSlice} from '@reduxjs/toolkit';

const main = createSlice({
    name: 'main',
    initialState:{
        tabs:[{id:'flats', value:'Квартиры'}, {id:'cottages', value:'Котеджы и усадьбы'}, {id:'baths', value:'Бани и сауны'},  {id:'cars', value:'Авто на прокат'}],

        cityOption:[{id:1,value:'Минск'},{id:2,value:'Гомель'},{id:3,value:'Брест'},{id:4,value:'Витебск'},{id:5,value:'Гродно'}, {id:6,value:'Могилев'}],
        cityCurrentValue:'',

        roomsOption: [{id:1,value:1},{id:2,value:2},{id:3,value:3},{id:4,value:4},{id:5,value:5}],
        roomsCurrentValue:'',

        metroOption:[{id:1,value:'Грушевка'},{id:2,value:'Восход'},{id:3,value:'Пушкинская'},{id:4,value:'Партизанская'}],
        metroCurrentValue:'',

        districtOption:[{id:1,value:'Шабаны'},{id:2,value:'Запад'},{id:3,value:'Восток'},{id:4,value:'Малиновка'}],
        districtCurrentValue: '',

        productsLimit: 6,
        currentPage: 1,

        sortedOptions:[{id:1, value:'Сначала дешевые'}, {id:2, value: 'Сначала дорогие'}],
        sortedValue:'',


        bedSumOptions:[{id:1, value:'1'}, {id:2, value:'2'}, {id:3, value:'3'}, {id:4, value:'4'}, {id:5, value:'5'}],
        bedSumCurrentValue:'',

        checkboxsMoreOption:[{id:1, value:'Газовая плита', bol:false},  {id:2, value:'Духовка', bol:false}, {id:3, value:'Кофеварка', bol:false}, {id:4, value:'Микроволновая печь ', bol:false}, {id:5, value:'Посуда', bol:false}, {id:6, value:'Посудомоечная машина', bol:false}],

        minCost:'',
        maxCost:'',


        productsLen: 0,
        sortedProducts:[],
        products: [],
    },
    reducers: {
        fetchProducts(state, action){
            state.products = action.payload
        },
        setBedSum(state, action){
            state.bedSumCurrentValue = action.payload
        },

        checkboxHandler(state, action){
            state.checkboxsMoreOption.forEach(obj =>{
                if(obj.value.includes(action.payload)){
                    obj.bol = !obj.bol

                }
            })
        },

        likeHandler(state, action){
            state.products.forEach(obj=>{
                Object.keys(obj).forEach(key=>{
                    if(key === action.payload.type){
                        obj[key].forEach(e=>{
                            if (e.id === action.payload.id){
                                e.like = !e.like
                            }
                        })
                    }
                })
            })
        },
        setCity(state, action){
            state.cityCurrentValue = action.payload

        },
        setRooms(state, action){
            state.roomsCurrentValue = action.payload
        },
        setMinCost(state, action){
            if(action.payload<0 || action.payload > 1000){
                return
            }
            state.minCost = action.payload
        },
        setMaxCost(state, action){
            if(action.payload<0 || action.payload > 1000){
                return
            }
            state.maxCost = action.payload
        },
        setDistrict(state, action){
            state.districtCurrentValue = action.payload
        },
        setMetro(state, action){
            state.metroCurrentValue = action.payload
        },
        setCurrentPage(state, action){
            state.currentPage = action.payload
            window.scrollTo(0, 0)
        },
        setSortedValue(state, action){
            state.sortedValue = action.payload
        },

        setLimit(state, action){
            state.productsLimit = action.payload
        },
        resetFilter(state){
            state.sortedValue = ''
            state.roomsCurrentValue = ''
            state.minCost = ''
            state.maxCost = ''
            state.districtCurrentValue = ''
            state.metroCurrentValue = ''
            state.bedSumCurrentValue = ''
            state.checkboxsMoreOption.forEach(e=>{
                e.bol = false
            })
        },

        mainFilter(state, action){
            // Условия для высчитывания стоимости
            while(state.minCost > state.maxCost ){
                state.maxCost = +state.maxCost + 100
            }
            if(state.minCost === ''){
                state.minCost=0
            }
            if(state.maxCost === ''){
                state.maxCost = 1000
            }


            //Основная фильтрация
            let sortedRes= []
            state.products.forEach(obj =>{
                if (action.payload){
                    Object.keys(obj).forEach(key =>{
                        if(action.payload === key){
                            obj[key].forEach(e=>{
                                if(!state.roomsCurrentValue){
                                    if(+e.cost >= state.minCost && +e.cost <= state.maxCost && e.city.includes(state.cityCurrentValue) && e.district.includes(state.districtCurrentValue) && e.metro.includes(state.metroCurrentValue) && e.people.includes(state.bedSumCurrentValue)){
                                        sortedRes.push(e)
                                    }
                                } else if(+e.cost >= state.minCost && +e.cost <= state.maxCost && e.city.includes(state.cityCurrentValue) && e.rooms===state.roomsCurrentValue && e.district.includes(state.districtCurrentValue) && e.metro.includes(state.metroCurrentValue) && e.people.includes(state.bedSumCurrentValue)){
                                    sortedRes.push(e)
                                }
                            })
                        }
                    })
                } else{
                    Object.keys(obj).forEach(key =>{
                        obj[key].forEach(e=>{
                            if(!state.roomsCurrentValue){
                                if(+e.cost >= state.minCost && +e.cost <= state.maxCost && e.city.includes(state.cityCurrentValue) && e.district.includes(state.districtCurrentValue) && e.metro.includes(state.metroCurrentValue) && e.people.includes(state.bedSumCurrentValue)){
                                    sortedRes.push(e)
                                }
                            } else if(+e.cost >= state.minCost && +e.cost <= state.maxCost && e.city.includes(state.cityCurrentValue) && e.rooms===state.roomsCurrentValue && e.district.includes(state.districtCurrentValue) && e.metro.includes(state.metroCurrentValue) && e.people.includes(state.bedSumCurrentValue)){
                                sortedRes.push(e)
                            }
                        })
                    })
                }
            })
            state.productsLen = sortedRes.length

            // Сортировка массива
            switch (state.sortedValue) {
            case 'Сначала дешевые':
                sortedRes.sort((a, b)=>{
                    return a.cost - b.cost
                })
                break
            case 'Сначала дорогие':
                sortedRes.sort((a, b)=>{
                    return b.cost - a.cost
                })
                break
            default:
                break
            }

            // Фильтрация по страницам
            let res = []
            let test = 0
            for (let i = 0; test < sortedRes.length; i++) {
                let obj = []
                for (let j = 0; j< state.productsLimit; j++) {
                    if(sortedRes[test]){
                        obj.push(sortedRes[test])
                        test++
                    }

                }
                res.push(obj)
            }
            state.sortedProducts = res
            state.currentPage = 1
        },

    },
})

export default main.reducer
export const {setCity, setRooms, setMinCost, setMaxCost, mainFilter, setDistrict, setMetro, resetFilter, setCurrentPage, setSortedValue, setLimit, likeHandler, setBedSum, checkboxHandler, fetchProducts} = main.actions
