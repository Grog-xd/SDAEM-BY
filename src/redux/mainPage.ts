import {createSlice, PayloadAction} from '@reduxjs/toolkit';

import {ProductProps} from '../types/types';


interface likeActionProps{
    id: number,
    type:string
}

const main = createSlice({
    name: 'main',
    initialState:{
        tabs:[{id:'flats', value:'Квартиры'}, {id:'cottages', value:'Котеджы и усадьбы'}, {id:'baths', value:'Бани и сауны'},  {id:'cars', value:'Авто на прокат'}],

        cityOption:[{id:1,value:'Минск'},{id:2,value:'Гомель'},{id:3,value:'Брест'},{id:4,value:'Витебск'},{id:5,value:'Гродно'}, {id:6,value:'Могилев'}],
        cityCurrentValue:'',

        roomsOption: [{id:1,value:1},{id:2,value:2},{id:3,value:3},{id:4,value:4},{id:5,value:5}],
        roomsCurrentValue:'',

        metroOption:[{id:1,value:'Грушевка'},{id:2,value:'Восход'},{id:3,value:'Пушкинская'},{id:4,value:'Партизанская'}],
        metroCurrentValue:'',

        districtOption:[{id:1,value:'Шабаны'},{id:2,value:'Запад'},{id:3,value:'Восток'},{id:4,value:'Малиновка'}],
        districtCurrentValue: '',

        productsLimit: 6,
        currentPage: 1,

        sortedOptions:[{id:1, value:'Сначала дешевые'}, {id:2, value: 'Сначала дорогие'}],
        sortedValue:'',


        bedSumOptions:[{id:1, value:'1'}, {id:2, value:'2'}, {id:3, value:'3'}, {id:4, value:'4'}, {id:5, value:'5'}],
        bedSumCurrentValue:'',

        checkboxsMoreOption:[{id:1, value:'Газовая плита', bol:false},  {id:2, value:'Духовка', bol:false}, {id:3, value:'Кофеварка', bol:false}, {id:4, value:'Микроволновая печь ', bol:false}, {id:5, value:'Посуда', bol:false}, {id:6, value:'Посудомоечная машина', bol:false}],

        minCost:0,
        maxCost:1000,


        productsLen: 0,
        sortedProducts:[],
        products: [],
    },
    reducers: {
        fetchProducts(state, action){
            state.products = action.payload
        },
        setBedSum(state, action:PayloadAction<string>){
            state.bedSumCurrentValue = action.payload
        },

        checkboxHandler(state, action:PayloadAction<string>){
            state.checkboxsMoreOption.forEach(obj =>{
                if(obj.value.includes(action.payload)){
                    obj.bol = !obj.bol

                }
            })
        },

        likeHandler(state, action:PayloadAction<likeActionProps>){
            state.products.forEach(obj=>{
                Object.keys(obj).forEach(key=>{
                    if(key === action.payload.type){
                        obj[key].forEach(e=>{
                            if (e.id === action.payload.id){
                                e.like = !e.like
                            }
                        })
                    }
                })
            })
        },
        setCity(state, action:PayloadAction<string>){
            state.cityCurrentValue = action.payload

        },
        setRooms(state, action:PayloadAction<string>){
            state.roomsCurrentValue = action.payload
        },
        setMinCost(state, action:PayloadAction<number>){
            if(action.payload<0 || action.payload > 1000){
                return
            }
            state.minCost = action.payload
        },
        setMaxCost(state, action:PayloadAction<number>){
            if(action.payload<0 || action.payload > 1000){
                return
            }
            state.maxCost = action.payload
        },
        setDistrict(state, action:PayloadAction<string>){
            state.districtCurrentValue = action.payload
        },
        setMetro(state, action:PayloadAction<string>){
            state.metroCurrentValue = action.payload
        },
        setCurrentPage(state, action:PayloadAction<number>){
            state.currentPage = action.payload
            window.scrollTo(0, 0)
        },
        setSortedValue(state, action:PayloadAction<string>){
            state.sortedValue = action.payload
        },

        setLimit(state, action:PayloadAction<number>){
            state.productsLimit = action.payload
        },
        resetFilter(state){
            state.sortedValue = ''
            state.roomsCurrentValue = ''
            state.minCost = undefined
            state.maxCost = undefined
            state.districtCurrentValue = ''
            state.metroCurrentValue = ''
            state.bedSumCurrentValue = ''
            state.checkboxsMoreOption.forEach(e=>{
                e.bol = false
            })
        },

        mainFilter(state, action:PayloadAction<ProductProps>){
            // Условия для высчитывания стоимости
            while(state.minCost > state.maxCost ){
                state.maxCost = state.maxCost + 100
            }
            if(state.minCost === undefined){
                state.minCost = 0
            }
            if(state.maxCost === undefined){
                state.maxCost = 1000
            }


            //Основная фильтрация
            let sortedRes= []
            state.products.forEach(obj =>{
                if (action.payload){
                    Object.keys(obj).forEach(key =>{
                        if(action.payload === key){
                            obj[key].forEach(e=>{
                                if(!state.roomsCurrentValue){
                                    if(+e.cost >= state.minCost && +e.cost <= state.maxCost && e.city.includes(state.cityCurrentValue) && e.district.includes(state.districtCurrentValue) && e.metro.includes(state.metroCurrentValue) && e.people.includes(state.bedSumCurrentValue)){
                                        sortedRes.push(e)
                                    }
                                } else if(+e.cost >= state.minCost && +e.cost <= state.maxCost && e.city.includes(state.cityCurrentValue) && e.rooms===state.roomsCurrentValue && e.district.includes(state.districtCurrentValue) && e.metro.includes(state.metroCurrentValue) && e.people.includes(state.bedSumCurrentValue)){
                                    sortedRes.push(e)
                                }
                            })
                        }
                    })
                } else{
                    Object.keys(obj).forEach(key =>{
                        obj[key].forEach(e=>{
                            if(!state.roomsCurrentValue){
                                if(+e.cost >= state.minCost && +e.cost <= state.maxCost && e.city.includes(state.cityCurrentValue) && e.district.includes(state.districtCurrentValue) && e.metro.includes(state.metroCurrentValue) && e.people.includes(state.bedSumCurrentValue)){
                                    sortedRes.push(e)
                                }
                            } else if(+e.cost >= state.minCost && +e.cost <= state.maxCost && e.city.includes(state.cityCurrentValue) && e.rooms===state.roomsCurrentValue && e.district.includes(state.districtCurrentValue) && e.metro.includes(state.metroCurrentValue) && e.people.includes(state.bedSumCurrentValue)){
                                sortedRes.push(e)
                            }
                        })
                    })
                }
            })
            state.productsLen = sortedRes?.length

            // Сортировка массива
            switch (state.sortedValue) {
            case 'Сначала дешевые':
                sortedRes.sort((a, b)=>{
                    return a.cost - b.cost
                })
                break
            case 'Сначала дорогие':
                sortedRes.sort((a, b)=>{
                    return b.cost - a.cost
                })
                break
            default:
                break
            }

            // Фильтрация по страницам
            let res = []
            let test = 0
            for (let i = 0; test < sortedRes?.length; i++) {
                let obj = []
                for (let j = 0; j< state.productsLimit; j++) {
                    if(sortedRes[test]){
                        obj.push(sortedRes[test])
                        test++
                    }

                }
                res.push(obj)
            }
            state.sortedProducts = res
            state.currentPage = 1
        },

    },
})

export default main.reducer
export const {setCity, setRooms, setMinCost, setMaxCost, mainFilter, setDistrict, setMetro, resetFilter, setCurrentPage, setSortedValue, setLimit, likeHandler, setBedSum, checkboxHandler, fetchProducts} = main.actions
