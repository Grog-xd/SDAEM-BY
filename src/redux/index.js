import {combineReducers} from "redux";
import {configureStore} from "@reduxjs/toolkit";
import mainPage from "./mainPage";
import newsPage from "./newsPage";
import loginPage from "./loginPage";

const rootReducer = combineReducers({
    news: newsPage,
    main:mainPage,
    login: loginPage
})

export const store = configureStore({
    reducer: rootReducer
})
