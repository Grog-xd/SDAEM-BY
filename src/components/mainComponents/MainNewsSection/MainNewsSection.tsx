import {FC, useEffect, useState} from 'react';
import axios from 'axios';
import {Link} from 'react-router-dom';
import {useDispatch, useSelector} from 'react-redux';

import {fetchNews, getNews} from '../../../redux/newsPage';
import {getPostUrl} from '../../../server';
import SvgArrowRight from '../../svg/SvgArrowRight.tsx';
import yellowPoints from '../../../assets/img/yellow-points.png';
import flatImg from '../../../assets/img/flat-img.png';

import {newProps, newsRedux} from '../../../types/types';

import classes from './MainNewsSection.module.scss';

const MainNewsSection:FC = () => {
    const dispatch = useDispatch()
    const news = useSelector((state:newsRedux)=> state.news.posts)
    const [newsArr, setNewsArr] = useState<newProps[]>([])
    const cityMainName = 'Минск'

    function newsFilter(){
        let res = []
        for (let i = 0; i < news?.length; i++) {
            if(res?.length === 5){
                break
            } else{
                res.push(news[i])
            }
        }
        setNewsArr(res)
    }

    useEffect(()=>{
        axios.get(`api/${getPostUrl}`)
            .then(response => response.data.posts)
            .then(response => dispatch(fetchNews(response)))
            .then(()=>dispatch(getNews('')))
            .catch(e => console.log(e.message))
    }, [])

    useEffect(()=>{
        newsFilter()
    }, [news])


    return (
        <section data-testid={'main-news-section'} className={classes.newsSection}>
            <div className={classes.textBlock}>
                <div className={classes.header}>
                    <strong>ЧТО ТАКОЕ SDAEM.BY</strong>
                    <p>Квартира на сутки в {cityMainName}е</p>
                </div>
                <div className={classes.body}>
                    <img className={classes.yellowPoints} src={yellowPoints} alt='Желтые точки'/>
                    <img className={classes.bodyImg} src={flatImg} alt='Изображение квартиры'/>
                    <div className={classes.bodyText}>
                        <p>
                            <b>Нужна квартира на сутки в {cityMainName}е?</b><br/>
                            На веб-сайте sdaem.by вас ждет масса выгодных предложений. Каталог насчитывает <b>более 500 квартир.</b> Благодаря удобной навигации вы быстро найдете подходящий вариант.
                        </p>
                        <p>
                            В каталоге представлены комфортабельные однокомнатные квартиры на сутки и квартиры с большим количеством комнат в разных районах города, с различной степенью удобства от дешевых до VIP
                            с джакузи.
                        </p>
                    </div>
                </div>
                <p className={classes.footer}>Чтобы снять квартиру на сутки в {cityMainName}е, вам достаточно определиться с выбором и связаться с владельцем для уточнения условий аренды и заключить договор. Заметим, на сайте представлены исключительно квартиры на сутки без посредников, что избавляет посетителей от необходимости взаимодействовать с агентствами, тратя свое время и деньги. Также пользователи сайта могут совершенно бесплатно размещать объявления о готовности сдать квартиру на сутки.    </p>
            </div>
            <div className={classes.newsBlock}>
                <p className={classes.title}>Новости</p>
                {
                    newsArr.map((post) =>
                        <div data-testid={'news-item'} key={post.id} className={classes.post}>
                            <Link to={`news/${post.id}`}>{post.title}</Link>
                            <p>{post.date}</p>
                        </div>,
                    )
                }
                <Link className={classes.toNewsLink} to={'/news'}>
                    Посмотреть все
                    <SvgArrowRight width={'9'} height={'16'} color={'#664EF9'}/>
                </Link>
            </div>
        </section>
    );
};

export default MainNewsSection;
