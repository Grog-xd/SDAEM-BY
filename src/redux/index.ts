import {combineReducers} from 'redux';
import {configureStore} from '@reduxjs/toolkit';
import {useDispatch} from 'react-redux';

import mainPage from './mainPage';
import newsPage from './newsPage';
import loginPage from './loginPage';

const rootReducer = combineReducers({
    news: newsPage,
    main:mainPage,
    login: loginPage,
})

export const store = configureStore({
    reducer: rootReducer,
})


export type AppDispatch = typeof store.dispatch
export const useAppDispatch: () => AppDispatch = useDispatch
