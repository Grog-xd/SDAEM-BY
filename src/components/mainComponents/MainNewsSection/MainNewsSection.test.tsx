import {render, screen} from '@testing-library/react';
import {Provider} from 'react-redux';

import axios, * as others from 'axios';
import {MemoryRouter} from 'react-router-dom';

import {store} from '../../../redux';
import postImg from '../../../assets/img/new-img.png';
import bigImg from '../../../assets/img/new-img-page.png';

import MainNewsSection from './MainNewsSection';


// jest.mock('axios')

describe('main news section tests', () =>{

    // let response
    //
    // beforeEach(() =>{
    //     response = {
    //         data:{
    //             posts:[
    //                 {id:1, img: postImg, bigImg:bigImg, title: 'test',  description: 'test description', body: 'Чем заняться в выходные? Когда нет безотлагательных домашних дел, а на улице хорошая погода, хочется уехать из города, чтобы сменить обстановку. Например, снять коттедж на сутки для семьи или большой компании друзей. А...', date: '14 января 2008' },
    //                 {id:2, img: postImg, bigImg:bigImg, title: 'test',  description: 'test description', body: 'Чем заняться в выходные? Когда нет безотлагательных домашних дел, а на улице хорошая погода, хочется уехать из города, чтобы сменить обстановку. Например, снять коттедж на сутки для семьи или большой компании друзей. А...', date: '14 января 2008' },
    //                 {id:3, img: postImg, bigImg:bigImg, title: 'test',  description: 'test description', body: 'Чем заняться в выходные? Когда нет безотлагательных домашних дел, а на улице хорошая погода, хочется уехать из города, чтобы сменить обстановку. Например, снять коттедж на сутки для семьи или большой компании друзей. А...', date: '14 января 2008' },
    //                 {id:4, img: postImg, bigImg:bigImg, title: 'test',  description: 'test description', body: 'Чем заняться в выходные? Когда нет безотлагательных домашних дел, а на улице хорошая погода, хочется уехать из города, чтобы сменить обстановку. Например, снять коттедж на сутки для семьи или большой компании друзей. А...', date: '14 января 2008' },
    //                 {id:5, img: postImg, bigImg:bigImg, title: 'test',  description: 'test description', body: 'Чем заняться в выходные? Когда нет безотлагательных домашних дел, а на улице хорошая погода, хочется уехать из города, чтобы сменить обстановку. Например, снять коттедж на сутки для семьи или большой компании друзей. А...', date: '14 января 2008' },
    //                 {id:6, img: postImg, bigImg:bigImg, title: 'test',  description: 'test description', body: 'Чем заняться в выходные? Когда нет безотлагательных домашних дел, а на улице хорошая погода, хочется уехать из города, чтобы сменить обстановку. Например, снять коттедж на сутки для семьи или большой компании друзей. А...', date: '14 января 2008' },
    //
    //             ]
    //         }
    //     }
    // })
    //
    // afterEach(() =>{
    //     jest.clearAllMocks()
    // })

    test('snapshot test', ()=>{
        render(
            <Provider store={store} >
                <MemoryRouter>
                    <MainNewsSection />
                </MemoryRouter>
            </Provider>,
        )
        const section = screen.getByTestId('main-news-section')
        expect(section).toMatchSnapshot();
    })

    // test('getting news test', async () =>{
    //     axios.get.mockReturnValue(response)
    //     render(
    //         <Provider store={store} >
    //             <MemoryRouter>
    //                 <MainNewsSection />
    //             </MemoryRouter>
    //         </Provider>
    //     )
    //     const posts = await screen.findAllByTestId('news-item')
    //     expect(posts.length).toBe(5)
    //     expect(axios.get).toBeCalledTimes(1)
    // })
})
