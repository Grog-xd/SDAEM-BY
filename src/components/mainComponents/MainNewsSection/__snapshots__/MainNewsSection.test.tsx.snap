// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main news section tests snapshot test 1`] = `
<section
  class="newsSection"
  data-testid="main-news-section"
>
  <div
    class="textBlock"
  >
    <div
      class="header"
    >
      <strong>
        ЧТО ТАКОЕ SDAEM.BY
      </strong>
      <p>
        Квартира на сутки в 
        Минск
        е
      </p>
    </div>
    <div
      class="body"
    >
      <img
        alt="Желтые точки"
        class="yellowPoints"
        src="yellow-points.png"
      />
      <img
        alt="Изображение квартиры"
        class="bodyImg"
        src="flat-img.png"
      />
      <div
        class="bodyText"
      >
        <p>
          <b>
            Нужна квартира на сутки в 
            Минск
            е?
          </b>
          <br />
          На веб-сайте sdaem.by вас ждет масса выгодных предложений. Каталог насчитывает 
          <b>
            более 500 квартир.
          </b>
           Благодаря удобной навигации вы быстро найдете подходящий вариант.
        </p>
        <p>
          В каталоге представлены комфортабельные однокомнатные квартиры на сутки и квартиры с большим количеством комнат в разных районах города, с различной степенью удобства от дешевых до VIP с джакузи.
        </p>
      </div>
    </div>
    <p
      class="footer"
    >
      Чтобы снять квартиру на сутки в 
      Минск
      е, вам достаточно определиться с выбором и связаться с владельцем для уточнения условий аренды и заключить договор. Заметим, на сайте представлены исключительно квартиры на сутки без посредников, что избавляет посетителей от необходимости взаимодействовать с агентствами, тратя свое время и деньги. Также пользователи сайта могут совершенно бесплатно размещать объявления о готовности сдать квартиру на сутки.    
    </p>
  </div>
  <div
    class="newsBlock"
  >
    <p
      class="title"
    >
      Новости
    </p>
    <a
      class="toNewsLink"
      href="/news"
    >
      Посмотреть все
      <svg
        fill="none"
        height="16"
        viewBox="0 0 9 16"
        width="9"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 14.8574L7.57143 8.28599L0.999999 1.71456"
          stroke="#664EF9"
        />
      </svg>
    </a>
  </div>
</section>
`;
